web/packages/teleport/src/Assist/Conversation/Message.tsx

```diff
@@ -196,39 +196,34 @@ export function Message(props: MessageProps) {
   );
 
   const authorIsTeleport = props.message.author === Author.Teleport;
-  const typing = authorIsTeleport && props.lastMessage && messages.streaming;
+  const showFooter =
+    !authorIsTeleport || (props.lastMessage ? !messages.streaming : true);
```

Comment:
I had an idea on how to simplify this, so I ran it through ChatGPT to confirm this. It turned out my idea was wrong, but ChatGPT tells me this can be simplified to 

```
!(authorIsTeleport && props.lastMessage && messages.streaming)
```

> The first line can be simplified by understanding the conditional logic and its outcome:
> 
> `!authorIsTeleport || (props.lastMessage ? !messages.streaming : true)`
> 
> Given that the only time you get `false` is when `authorIsTeleport` is `true` and `props.lastMessage` exists and `messages.streaming` is also `true`, you can write it in a simplified form:
> 
> `!(authorIsTeleport && props.lastMessage && messages.streaming)`
> 
> The logic is equivalent because the negation operator (`!`) negates the entire expression inside the parentheses.

I think that's correct but just to be sure I asked it to generate code to verify this in a brute-force fashion:

<details>

```typescript
// Define all possible boolean values
let booleanValues = [true, false, null, undefined];

let allEqual = true;  // Assume all results are equal initially

// Run through all possible combinations
for (let authorIsTeleport of booleanValues) {
    for (let lastMessage of booleanValues) {
        for (let streaming of booleanValues) {
            let props = { lastMessage };
            let messages = { streaming };
            
            let result1 = !authorIsTeleport || (props.lastMessage ? !messages.streaming : true);
            let result2 = !(authorIsTeleport && props.lastMessage && messages.streaming);

            // If we find a pair of results that aren't equal, set allEqual to false
            if (result1 !== result2) {
                allEqual = false;
                console.log(`Unequal results found with authorIsTeleport=${authorIsTeleport}, lastMessage=${lastMessage}, streaming=${streaming}`);
                break;
            }
        }
        if (!allEqual) break;  // Break outer loops if a difference has been found
    }
    if (!allEqual) break;  // Break outer loops if a difference has been found
}

console.log(`Are all results equal? ${allEqual}`);
```
</details>

Reply:
Cheers for this, that's much nicer.